### PR TITLE
Convenient Use of Function Parameters

### PR DESCRIPTION
**Pull request by wistuba**
Addresses Part 1 of #115: Users can use arbitrary argument names for the functions and parameters are automatically forwarded based on the function interface.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/renate/cli/parsing_functions.py [hunk 1]
```diff
@@ -493,32 +638,96 @@ def get_transforms_kwargs(
...
+    "ER": _add_experience_replay_arguments,
+    "DER": _add_dark_experience_replay_arguments,
+    "POD-ER": _add_pod_experience_replay_arguments,
+    "CLS-ER": _add_cls_experience_replay_arguments,
+    "Super-ER": _add_super_experience_replay_arguments,
+    "GDumb": _add_gdumb_arguments,
+    "Joint": _add_joint_arguments,
+    "FineTuning": _add_finetuning_arguments,
+    "RD": _add_rd_learner_arguments,
+    "OfflineER": _add_replay_learner_arguments,
```
Comment (610v4nn1) [inline src/renate/cli/parsing_functions.py:732]: Do we really need to maintain these functions? can we remove them in favor of an automatic detection of the arguments of each MU with get_function_args (or similar)?

**Comment by 610v4nn1** [comment]
It would be good to explain the usage of the arguments in the documentation

File: src/renate/cli/parsing_functions.py (hunk 1 above)
Comment (wistuba) [reply src/renate/cli/parsing_functions.py:740]: These functions extend the arguments parsed by argsparse. I don't see how we can remove this.

**Comment by wistuba** [comment]
> It would be good to explain the usage of the arguments in the documentation

This is only the first part of a larger change that I break down into self-contained features to be better consumed by the reviewers. Explaining the current state won't make too much sense since it will change a bit. I'll add it with the next change.

File: src/renate/cli/parsing_functions.py (hunk 1 above)
Comment (610v4nn1) [reply src/renate/cli/parsing_functions.py]: As discussed offline, let's open a separate issue for this

**Pull request merged**